core/modules/css/KvInterfaceImplMem.cc

```diff
@@ -194,12 +206,35 @@ KvInterfaceImplMem::getChildren(string const& key) {
 void
 KvInterfaceImplMem::deleteKey(string const& key) {
     LOGF_DEBUG("deleteKey(%1%)." % key);
+
+    if (_readOnly) {
+        throw ReadonlyCss();
+    }
+
     if ( ! exists(key) ) {
         throw NoSuchKey(key);
     }
     _kvMap.erase(key);
 }
 
+std::string KvInterfaceImplMem::dumpKV() {
+    std::string result;
+    for (auto& pair: _kvMap) {
+        if (not result.empty()) result += '\n';
```

Comment:
It's jarring (this is the only C++ code base where I've seen not used), but I can't claim it's harder to read.